front/src/modules/timesStops/hooks/useOutputTableData.ts

```diff
@@ -4,154 +4,124 @@ import { keyBy } from 'lodash';
-  );
-  const pathStepsByUic = keyBy(
-    pathStepsWithOpPointIndices,
-    (pathStep) => `${pathStep.uic}-${pathStep.secondary_code}`
-  );
-
-  const outputTableData = useMemo(() => {
-    const pathStepRows = pathStepsWithPositionOnPath.map((pathStep) => {
-      const schedule = scheduleByAt[pathStep.id];
-      const scheduleData = computeScheduleData(schedule);
-      return { ...pathStep, ...formatScheduleData(scheduleData) };
-    });
+  const scheduleByAt: Record<string, ScheduleEntry> = keyBy(selectedTrainSchedule?.schedule, 'at');
 
-    const suggestedOpRows = suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
-      const opPoint = operationalPoints.find((op) => op.id === sugOpPoint.opId);
-      if (!opPoint) {
-        return sugOpPoint;
-      }
-      const nextOpPoint = findNextScheduledOpPoint(
-        operationalPoints,
-        pathStepsWithOpPointIndices,
-        sugOpIndex
+  const pathStepRows = useMemo(() => {
+    if (!path || !selectedTrainSchedule || !trainSummary?.pathItemTimes) return [];
+
+    const startDatetime = new Date(selectedTrainSchedule.start_time);
+
+    const result: (TimeStopsRow & { positionOnPath: number })[] = [];
+    let previousTime = new Date(selectedTrainSchedule.start_time);
+
+    for (let index = 0; index < selectedTrainSchedule.path.length; index += 1) {
+      const pathStep = selectedTrainSchedule.path[index];
+      const schedule: ScheduleEntry | undefined = scheduleByAt[pathStep.id];
+
+      const computedArrival = new Date(
+        trainSummary.pathItemTimes.final[index] + startDatetime.getTime()
+      );
+
+      const { stopFor, shortSlipDistance, onStopSignal, calculatedDeparture } = formatSchedule(
+        computedArrival,
+        schedule
+      );
+      const { theoreticalMargin, theoreticalMarginSeconds, calculatedMargin, diffMargins } =
+        computeMargins(selectedTrainSchedule, index, trainSummary.pathItemTimes);
+
+      const inputs = computeInputDatetimes(startDatetime, previousTime, schedule, {
+        isDeparture: index === 0,
+      });
+      const { arrival, departure } = inputs;
+      previousTime = inputs.prev;
+
+      result.push({
+        opId: pathStep.id,
+        name: t('waypoint', { id: pathStep.id }),
+        ch: undefined,
+        isWaypoint: true,
+
+        arrival,
+        departure,
+        stopFor,
+        onStopSignal,
+        shortSlipDistance,
+        theoreticalMargin,
+
+        theoreticalMarginSeconds,
+        calculatedMargin,
+        diffMargins,
+        calculatedArrival: dateToHHMM(computedArrival, { withSeconds: true }),
+        calculatedDeparture,
+        positionOnPath: path.path_item_positions[index],
+      });
+    }
+
+    return result;
+  }, [selectedTrainSchedule, path, trainSummary?.pathItemTimes]);
+
+  const rows: TimeStopsRow[] = useMemo(() => {
+    if (!pathProperties || !selectedTrainSchedule) return [];
+
+    const startDatetime = new Date(selectedTrainSchedule.start_time);
```

Comment:
We could define this only once at the top of the hook (won't to check for `undefined` if you moved the Loader before this hook call)